src/error.rs

```diff
@@ -31,7 +33,10 @@ impl Error {
     ///
     /// If the argument implements std::error::Error, prefer `Error::new`
     /// instead which preserves the underlying error's cause chain and
-    /// backtrace. If the argument may or may not implement std::error::Error
+    /// backtrace, unless in a `no_std` crate (in which case `Error::new`
+    /// is unavailable).
```

Comment:
Is this distinction necessary? How could your error implement std::error::Error in a no-std crate?